src/shogun/io/TBOutputFormat.cpp

```diff
@@ -106,6 +107,13 @@ tensorflow::Event TBOutputFormat::convert_scalar(
 		    "Unsupported type %s", value.first.get_value().type_info().name());
 	}
 
+	auto write_summary = [&summaryValue=summaryValue, &value](auto type) {
+		summaryValue->set_simple_value(
+		    any_cast<decltype(type)>(value.first.get_value()));
+	};
+
+	sg_for_each_type(value.first.get_value(), sg_all_types, write_summary);
```

Comment:
I think you might kill me, but maybe "dispatch" should be part of the function name? `sg_any_dispatch` comes to my mind now.... but not sure what is better. What are your toughts?

Reply:
Yes that sounds good! Was just waiting for a final decision before changing the name!

Reply:
lol ... let's hope this is the last :D